src/github.com/t3chguy/riot-static/data.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Notes
+// Reads should be done on clones
+// Treat Rooms as if Immutable
+// ONLY MUTATE DURING A FULL LOCK
+
+package main
+
+import (
+	"sync"
+)
+
+type DataStore struct {
+	sync.RWMutex
+	roomList []*Room
+	roomMap  map[string]*Room
+}
+
+//func (data *DataStore) GetRoomPointer(roomID string) *Room {
+//	data.RLock()
+//	defer data.RUnlock()
+//	return data.roomMap[roomID]
+//}
+
+func (data *DataStore) GetRoom(roomID string) (room Room, exists bool) {
+	data.RLock()
+	roomPointer := data.roomMap[roomID]
+	data.RUnlock()
+
+	if roomPointer == nil {
+		return Room{}, false
+	}
+	exists = true
+
+	data.RLock()
+	expired := roomPointer.Cached.CheckExpired()
+	data.RUnlock()
```

Comment:
Right, I was going to let you off on this, but this has convinced me not to. You need to tell the reader what your mutex is supposed to be protecting (a comment or an obvious name e.g. `roomMapMutex`).

Why the fuss you may ask? Well, without it, I cannot verify if you're actually protecting what you think you are protecting. I thought the embedded mutex was protecting the `roomMap`, but then why are you taking a lock out here when you don't access it?

Locks are hard enough to get right as it is, don't make it harder for yourself or me.